build_modules/README.md

```diff
@@ -32,3 +32,70 @@ building your Dart code.
 
 [`build_web_compilers`]: https://pub.dartlang.org/packages/build_web_compilers
 [`build_vm_compilers`]: https://pub.dartlang.org/packages/build_vm_compilers
+
+## Module algorithms
+
+The overall process for emitting modules follows these steps:
+
+1. Emit a `.module.library` asset for every `.dart` library containing a
+   description of the directives (imports, exports, and parts), a list of the
+   `dart:` imports used, and a determination of whether the library is an
+   "entrypoint". Entrypoint libraries are either those which are likely to be
+   directly imported (they are under `lib/` but not `lib/src/`) or which have a
+   `main`. Only the libraries that can be imported (they aren't part files) will
+   have an output. This step is mainly separated out for better invalidation
+   behavior since the output will change less frequently than the code. The
+   outputs from this step are non differentiated by platform.
+1. Emit package level `.meta_module` assets. This is an aggregation of the
```

Comment:
2. (etc)

Reply:
Discussed offline.

Markdown will render this with the correct numbers, using `1.` for all of them means if we add new entries we don't need to manually renumber them ourselves and keeps the diff cleaner.

Given that the number of steps is unlikely to change here, and is pretty small, I'm fine doing a manual numbering since it does look nicer in the source.